contribs/emissions/src/main/java/org/matsim/contrib/emissions/ColdEmissionAnalysisModule.java

```diff
@@ -319,7 +319,19 @@ private HbefaColdEmissionFactor getEmissionsFactor(Tuple<HbefaVehicleCategory, H
 					logger.info(Gbl.FUTURE_SUPPRESSED);
 					detailedReadingInfoCnt++;
 				}
-				if (this.detailedHbefaColdTable.get(efkey) != null) {
+				/* The `this.detailedHbefaColdTable.get(efkey)` call may result in a NullPointerException
```

Comment:
You can get a `NullPointerException` if `this.detailedHbefaColdTable` is `null`. If this is the case the loading of the tables in the 'EmissionModule' https://github.com/matsim-org/matsim-libs/blob/45776ed01dfec9765541b5c89cf133dcbd5fc9f4/contribs/emissions/src/main/java/org/matsim/contrib/emissions/EmissionModule.java#L122 doesn't work properly, or you are trying something which is not covered by your `DetailedVsAverageLookupBehavior`.

The second option where this Exception could come from is during the lookup in the hashmap. The `HbefaVehicleAttributes` override `hashCode` https://github.com/matsim-org/matsim-libs/blob/45776ed01dfec9765541b5c89cf133dcbd5fc9f4/contribs/emissions/src/main/java/org/matsim/contrib/emissions/HbefaVehicleAttributes.java#L95 which is called during the `Map::get` call. Instead of wrapping this in a `try-catch` block, the `HbefaVehicleAttributes.sizeClass` should have a reasonable default. The current default is `average`.https://github.com/matsim-org/matsim-libs/blob/45776ed01dfec9765541b5c89cf133dcbd5fc9f4/contribs/emissions/src/main/java/org/matsim/contrib/emissions/HbefaVehicleAttributes.java#L30 

This means some other code intentionally sets this to `null`. If that's what you want, I'd rather suggest that you fix the `hashCode` method of `HbefaVehicleAttributes`. 

My personal preference is to set this value to something valid instead, because this avoids having null checks everywhere in the code. Maybe the current default would already be sufficent?

Reply:
Noted, thanks @Janekdererste. The problem does seem to be on the `Map::get` call. So the best place to fix it would be in the `hashCode` method when I can make some time. For some reason the `average` seems to be overridden... and that's what I need to figure out. I would prefer that it is rather `not specified` by default. But that is, again, an HBEFA 4.x format that is not backward compatible.

Reply:
> because this avoids having null checks everywhere in the code.

That makes sense, yes.